### PR TITLE
add internal sol to cses book shop

### PR DESCRIPTION
**Pull request by TruaShamu**

### CODE REVIEW

File: content/4_Gold/Knapsack_DP.problems.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
       "tags": ["Knapsack"],
       "solutionMetadata": {
         "kind": "link",
```
Comment (maggie-j-liu) [inline content/4_Gold/Knapsack_DP.problems.json:63]:
```suggestion
```

File: solutions/CSES Book Shop.mdx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		int[] pages = new int[N];
+		st = new StringTokenizer(b.readLine());
+		for (int i = 0; i < N; i++) {
+			pages[i] = Integer.parseInt(st.nextToken());
+		}
+
+		// dp[i][j] is how many pages you have from the first i books,
+		// after spending j money.
+		int[][] dp = new int[N + 5][X + 5];
+		for (int book = 1; book <= N; book++) {
```
Comment (dongliu0426) [inline solutions/CSES Book Shop.mdx:48]: just loop through from [0, N - 1]

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

File: solutions/CSES Book Shop.mdx (hunk 2 above)
Comment (TruaShamu) [reply solutions/CSES Book Shop.mdx]: Ok, I changed it.

File: solutions/CSES Book Shop.mdx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+		int[] pages = new int[N];
+		st = new StringTokenizer(b.readLine());
+		for (int i = 0; i < N; i++) {
+			pages[i] = Integer.parseInt(st.nextToken());
+		}
+ 
+		// dp[i][j] is how many pages you have from the first i books,
+		// after spending j money.
+		int[][] dp = new int[N + 5][X + 5];
+		for (int book = 0; book <= N-1; book++) {
```
Comment (dongliu0426) [inline solutions/CSES Book Shop.mdx:48]:
```suggestion
		for (int book = 0; book < N; book++) {
```

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**